src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -1127,75 +1131,135 @@ void verify(ModuleSymbol module)
+                                Driver.Count++;
 
-            Driver.Result = Driver.Count - tests;
-        }
-        finally
-        {
-            //When test complete, set the flag.
-            Driver.CompletedSignal.Set();
-        }
-    }
-}
-public class MyTaskAwaiter<T> : INotifyCompletion
-{
-    public void OnCompleted(Action continuationAction)
-    {
-    }
+                            Driver.Result = Driver.Count - tests;
+                        }
+                        finally
+                        {
+                            //When test complete, set the flag.
+                            Driver.CompletedSignal.Set();
+                        }
+                    }
+                }
+                public class MyTaskAwaiter<T> : {{(useCritical ? "ICriticalNotifyCompletion" : "INotifyCompletion")}}
+                {
+                    public void OnCompleted(Action continuationAction)
+                    {
+                    }
 
-    public T GetResult()
-    {
-        return default(T);
-    }
+                    public void UnsafeOnCompleted(Action continuationAction)
+                    {
+                    }
 
-    public bool IsCompleted { get { return true; } }
-}
-//-------------------------------------
+                    public T GetResult()
+                    {
+                        return default(T);
+                    }
 
-class Driver
-{
-    public static int Result = -1;
-    public static int Count = 0;
-    public static AutoResetEvent CompletedSignal = new AutoResetEvent(false);
-    static void Main()
-    {
-        new MyTask<int>().Run<MyTask<int>>(new MyTask<int>());
+                    public bool IsCompleted { get { return true; } }
+                }
+                //-------------------------------------
 
-        CompletedSignal.WaitOne();
+                class Driver
+                {
+                    public static int Result = -1;
+                    public static int Count = 0;
+                    public static AutoResetEvent CompletedSignal = new AutoResetEvent(false);
+                    static void Main()
+                    {
+                        new MyTask<int>().Run<MyTask<int>>(new MyTask<int>());
+
+                        CompletedSignal.WaitOne();
+
+                        // 0 - success
+                        // 1 - failed (test completed)
+                        // -1 - failed (test incomplete - deadlock, etc)
+                        Console.WriteLine(Driver.Result);
+                    }
+                }
+                """;
 
-        // 0 - success
-        // 1 - failed (test completed)
-        // -1 - failed (test incomplete - deadlock, etc)
-        Console.WriteLine(Driver.Result);
-    }
-}";
             CompileAndVerify(source, "0");
+
+            var comp = CreateCompilation([source, RuntimeAsyncAwaitHelpers], targetFramework: TargetFramework.Net90, parseOptions: WithRuntimeAsync(TestOptions.RegularPreview));
```

Comment:
Should we pass an `ExpectedOutput()` call here so that it lights up once a supported runtime for testing is in place?